heat/core/tests/test_rounding.py

```diff
@@ -162,6 +164,139 @@ def test_floor(self):
+    #      self.assertEqual(float64_modf_distrbd[1].dtype, ht.float64)
+    #
+    #      self.assertAlmostEqual(float64_modf_distrbd[0].numpy().all(), comparison[0].all())
+    #      self.assertAlmostEqual(float64_modf_distrbd[1].numpy().all(), comparison[1].all())
+    #
+    #      self.assertAlmostEqual(ht.all(float64_modf_distrbd[0]), ht.all(ht.array(comparison[0])))
+    #      self.assertAlmostEqual(ht.all(float64_modf_distrbd[1]), ht.all(ht.array(comparison[1])))
+
+    def test_round(self):
+        size = ht.communication.MPI_WORLD.size
+        start, end = -5.0, 5.0
+        step = (end - start) / (2 * size)
+        comparison = torch.arange(start, end, step, dtype=torch.float32).round()
+
+        # exponential of float32
+        float32_tensor = ht.array(comparison, dtype=ht.float32)
+        float32_round = float32_tensor.round()
+        self.assertIsInstance(float32_round, ht.DNDarray)
+        self.assertEqual(float32_round.dtype, ht.float32)
+        self.assertEqual(float32_round.dtype, ht.float32)
+        BasicTest.assert_array_equal(self, float32_round, comparison)
+
+        # exponential of float64
+        comparison = torch.arange(start, end, step, dtype=torch.float64).round()
+        float64_tensor = ht.array(comparison, dtype=ht.float64)
+        float64_round = float64_tensor.round()
+        self.assertIsInstance(float64_round, ht.DNDarray)
+        self.assertEqual(float64_round.dtype, ht.float64)
+        self.assertEqual(float64_round.dtype, ht.float64)
+        BasicTest.assert_array_equal(self, float64_round, comparison)
+
+        # check exceptions
+        with self.assertRaises(TypeError):
+            ht.round([0, 1, 2, 3])
+        with self.assertRaises(TypeError):
+            ht.round(object())
+        with self.assertRaises(TypeError):
+            ht.round(float32_tensor, 1, 1)
+
+        # with split tensors
+
+        # exponential of float32
+        comparison = torch.arange(start, end, step, dtype=torch.float32)  # .round()
+        float32_tensor_distrbd = ht.array(comparison, split=0)
+        comparison = comparison.round()
+        float32_round_distrbd = float32_tensor_distrbd.round()
+        self.assertIsInstance(float32_round_distrbd, ht.DNDarray)
+        self.assertEqual(float32_round_distrbd.dtype, ht.float32)
+        BasicTest.assert_array_equal(self, float32_round_distrbd, comparison)
+
+        # exponential of float64
+        comparison = torch.arange(start, end, step, dtype=torch.float64)  # .round()
+        float64_tensor_distrbd = ht.array(comparison, split=0)
+        comparison = comparison.round()
+        float64_round_distrbd = float64_tensor_distrbd.round()
+        self.assertIsInstance(float64_round_distrbd, ht.DNDarray)
+        self.assertEqual(float64_round_distrbd.dtype, ht.float64)
+        self.assertEqual(float64_round_distrbd.dtype, ht.float64)
+        BasicTest.assert_array_equal(self, float64_round_distrbd, comparison)
```

Comment:
It's better to extend the BasicTest class and then use `self.assert_array_equal`. This way self does not need to be passed additionally.